lib/saluki-components/src/transforms/host_tags/mod.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    num::NonZeroUsize,
+    time::{Duration, Instant},
+};
+
+use async_trait::async_trait;
+use bytesize::ByteSize;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_context::{ContextResolver, ContextResolverBuilder};
+use saluki_core::{components::transforms::*, topology::interconnect::FixedSizeEventBuffer};
+use saluki_env::helpers::remote_agent::RemoteAgentClient;
+use saluki_error::{generic_error, GenericError};
+use saluki_event::metric::Metric;
+
+/// Host Tags synchronous transform.
```

Comment:
Nit: should have a secondary paragraph with a small description about adding host tags for a fixed period of time to help with new hosts, etc etc.